fastly/helpers.go

```diff
@@ -31,3 +33,22 @@ func NullString(v string) *string {
 	}
 	return &v
 }
+
+// ToSafeURL produces a safe (no path traversal, no unsafe characters) URL
+// from the path components passed in.
+func ToSafeURL(unsafeComponents ...string) string {
+	safeComponents := make([]string, len(unsafeComponents))
+
+	for i := range unsafeComponents {
+		if component := unsafeComponents[i]; component != ".." {
```

Comment:
Coming to this not knowing anything, I might wonder what would happen when we skip over this index. Probably best to comment that an empty component is ignored and the resulting path only contains single slashes between `safeComponents`.

Reply:
I so see the test coverage, so that helps.

Reply:
Comment added.